keps/sig-node/4569-cgroup-v1-deprecation/README.md

```diff
@@ -0,0 +1,685 @@
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Deprecate the support for cgroup v1 in Kubernetes, aligning with the industry's move towards cgroup v2 as the default for Linux kernel resource management and isolation.
+
+## Motivation
+
+cgroup v2 offers significant improvements over cgroup v1 in terms of feature set, uniformity of interface, and scalability. With major Linux distributions and container runtimes supporting cgroup v2, it is timely for Kubernetes to deprecate cgroup v1 support to streamline maintenance and encourage the adoption of newer, more secure, and more performant technologies.
+
+
+### Goals
+
+- Deprecation of cgroup v1 support in Kubernetes.
+- Provide clear migration paths for users running Kubernetes on systems using cgroup v1.
+- Ensure test coverage
+
+### Non-Goals
+
+Removing cgroup v1 support immediately. The deprecation process will be gradual, allowing users ample time to migrate.
+
+## Proposal
+
+The proposal outlines a plan to deprecate cgroup v1 support in Kubernetes, encouraging the community and users to transition to cgroup v2.
+
+### Risks and Mitigations
```

Comment:
Should we have a section here to state that many cloud vendors are defaulting to cgroupv2? Or at least mention that all the cloud vendors support cgroupv2 now?

Openshift has GA cgroupv2 in 4.14.

Reply:
I am not sure if k8s should make vendor specific references. Our reasons to deprecate v1 are, v2 is better and v1 support from the things k8s depends up like systemd, kernel etc is disappearing.

Reply:
The case for posting it is to show that we are deprecating this and cgroupv2 is heavily supported in the popular environments. I think we are ready to do that but wasn't sure if we should mention that many of the hyperscalers default settings are cgroup v2.

I'm happy to be wrong on this one if you think its not necessary.

Reply:
Many of the users on the public clouds are still on cgroupv1 though.